modules/javafx.graphics/src/main/java/javafx/scene/LightBase.java

```diff
@@ -56,33 +56,112 @@
+ *     <td style="text-align:center">&#10003;</td>
+ *   </tr>
+ * </table>
+ *
+ * * Supports spotlight attenuation factors as described in its class docs.
+ * </div>
+ * <p>
+ * An application cannot add its own light types. Extending {@code LightBase} directly may lead to an
+ * {@code UnsupportedOperationException} being thrown.
+ * <p>
+ * All light types are not affected by scaling and shearing transforms.
+ *
+ * <h2><a id="Color">Color</a></h2>
+ * A light source produces a light of a single color. The appearance of an object illuminated by the light is a result
+ * of the color of the light and the material the object is made of. For example, for a blue light, a white object will
+ * appear blue, a black object will appear black, and a (255, 0, 255)-colored object will appear blue, as the light
+ * "selects" only the blue component of the material color. The mathematical definition of the material-light
+ * interaction is available in {@link javafx.scene.paint.PhongMaterial PhongMaterial}.
+ * <p>
+ * Notes about the current implementation:
+ * <ol>
+ *   <li> A black colored light is ignored since its contribution is 0.
+ *   <li> The transparency (alpha) component of a light is ignored.
+ * </ol>
+ * There are no guarantee that these behaviors will not change.
+ *
+ * <h2><a id="Scopes">Scopes</a></h2>
+ * A light has a {@code scope} list and an {@code exclusionScope} list that define which nodes are illuminated by it and
+ * which aren't. A node can exist in only one of the lists: if it is added to one, it is silently removed from the
+ * other. If a node does not exist in any list, it inherits its affected state from its parent, recursively. An
+ * exception to this is that a light with an empty {@code scope} affects all nodes in its scene/subscene implicitly
+ * (except for those in its {@code exlusionScope}) as if the root of the scene is in the {@code scope}.
  * <p>
- * A node can exist in only one of the lists, if it is added to one, it is silently removed from the other. If a node
- * does not exist in any list, it inherits its affected state from its parent, recursively. An exception to this is that
- * a light with an empty {@code scope} affects all nodes in its scene/subscene implicitly (except for those in its
- * {@code exlusionScope}) as if the root of the scene is in the {@code scope}. <br>
  * The {@code exlusionScope} is useful only for nodes that would otherwise be in scope of the light. Excluding a node is
  * a convenient alternative to traversing the scenegraph hierarchy and adding all of the other nodes to the light's
- * scope. Instead, the scope can remain wide and specific nodes can be excluded.
+ * scope. Instead, the scope can remain wide, and specific nodes can be excluded with the exclusion scope.
  *
+ * <h2><a id="Direction">Direction</a></h2>
+ * The direction the light is facing, defined by the {@code direction} vector property of the light. The light's
```

Comment:
The direction of the light is defined by the {@code direction} vector property of the light